### DISCUSSION TITLE
using MiEx with maya and renderman

### DISCUSSION THREAD

**Discussion by RonoBlag** [BramStoutProductions/MiEx#88] [Q&A]
hello, I have a problem when using MiEx and renderman. After creating my usd scene and starting the render with arnold everything appear, even if I have some issue with the way texture appear (their are flat, no pixel, like their where blended), but when I render with renderman I have nothing at all in my IPR.

Do you have an idea of what could cause that ?

**Comment by BramStout** [comment]
When creating USD exports, MiEx also needs to create the materials. However, every render engine has their own shading nodes and they don't all necessarily support the same ones. So, MiEx comes with a material templating system with which it creates the materials. This allows anyone to create material templates for whatever render engine they use and for whatever look they want. You can read more about material templates [here.](https://github.com/BramStoutProductions/MiEx/wiki/4.-Resource-Packs#material-templates)

This means that in order for exports to render out properly, you need the right material templates. The repository has [a list of example resource packs](https://github.com/BramStoutProductions/MiEx/tree/main/extras/example_resource_packs) that contain material templates for a few different render engines. The resource packs that end with "ForMaya" are meant to be used together with the Maya Import script. You can read more about that [here.](https://github.com/BramStoutProductions/MiEx/wiki/8.-Importing-into-DCCs#autodesk-maya) There is also another resource pack called "Renderman" which contains material templates with pure Renderman USD shading nodes. That can be used when using Renderman via Hydra. You should be able to use these resource packs directly, but they are example resource packs. Feel free to edit them to make your own.

MiEx does export out some preview vertex colours that can be shown when materials are either disabled or no supported materials are available. That's what you're seeing when rendering with Arnold. Renderman on the other hand just doesn't render out an object if it doesn't have a material assigned that it supports.
